src/queryExecutionContext/EndpointComponent/OffsetLimitEndpointComponent.ts

```diff
@@ -1,53 +1,49 @@
 import { Response } from "../../request";
+import { ExecutionContext } from "../ExecutionContext";
 import { getInitialHeader } from "../headerUtils";
-import { IExecutionContext } from "../IExecutionContext";
-import { IEndpointComponent } from "./IEndpointComponent";
 
 /** @hidden */
-export class TopEndpointComponent implements IEndpointComponent {
+export class OffsetLimitEndpointComponent implements ExecutionContext {
   /**
    * Represents an endpoint in handling top query. It only returns as many results as top arg specified.
    * @constructor TopEndpointComponent
    * @param { object } executionContext - Underlying Execution Context
    * @ignore
    */
-  constructor(private executionContext: IExecutionContext, private topCount: number) {}
+  constructor(private executionContext: ExecutionContext, private limit: number, private offset: number) {}
```

Comment:
Can we switch the argument order, so that it's offset, then limit (to match the query operation).